dask/order.py

```diff
@@ -381,6 +388,42 @@ def process_runnables() -> None:
     # writing, the most expensive part of ordering is the prep work (mostly
     # connected roots + sort_key) which can be reused for multiple orderings.
 
+    def get_target(longest_path: bool = False) -> Key:
+        # Some topologies benefit if the node with the most dependencies
+        # is used as first choice, others benefit from the opposite.
+        candidates = leaf_nodes
+        skey: Callable = sort_key
+
+        preferred_candidates = set()
+        if runnable_hull:
+            skey = lambda k: (num_needed[k], sort_key(k))
+            preferred_candidates = runnable_hull & candidates
+        if not preferred_candidates and reachable_hull:
+            skey = lambda k: (num_needed[k], sort_key(k))
+            preferred_candidates = reachable_hull & candidates
+
+        if reachable_hull or runnable_hull:
```

Comment:
I think we don't want to use runnable_hull here. In fact, reviewing the code more broadly, with 
https://github.com/dask/dask/pull/11305 it is only used within `process_runnables` so it might make sense to rename it to `seen_by_process_runnables` and avoid some confusion.

You are also not using it in the body of the if clause which likely indicates that this should go away

```suggestion
        if reachable_hull:
```

Reply:
Yeah that makes sense